packages/addons/src/plugins/maxlengthCountdown.ts

```diff
@@ -0,0 +1,121 @@
+): FormKitPlugin {
+
+  const maxlengthCountdownPlugin = (node: FormKitNode) => {
+    node.addProps(['maxlengthCountdown'])
+    node.addProps(['maxlengthRemainingText'])
+    node.addProps(['maxlengthStringCount'])
+
+    let usePlugin = MaxlengthCountdownOptions?.useAsDefault === true
+
+    let allowedTypes = MaxlengthCountdownOptions?.fieldTypes || ['textarea']
+    if (typeof allowedTypes === 'string') allowedTypes = [allowedTypes]
+    if (!allowedTypes.includes(node.props.type)) {
+      usePlugin = false
+    }
+
+    // ALLOW OVERRIDES FOR NON-DEFAULT FIELDS IF EXPLICITLY SET
+    if (undefine(node.props.maxlengthCountdown) ||
+      node.props.maxlengthCountdown === 'true' ||
+      node.props.maxlengthCountdown === true) {
+      usePlugin = true
+    }
+    
+    if (usePlugin) {
+      node.on('created', () => {
+        if (!node.props || !node.props.definition) return
+        if (node.props.attrs.maxlength === undefined) return
+
+        if (node.props.attrs.maxlength && parseInt(node.props.attrs.maxlength) > 0) {
+          const inputDefinition = clone(node.props.definition)
+          const originalSchema = inputDefinition.schema
+          if (typeof originalSchema !== 'function') return
+
+          const higherOrderSchema = (
+            extensions: Record<
+              string,
+              Partial<FormKitSchemaNode> | FormKitSchemaCondition
+            >
+          ) => {
+            extensions.suffix = {
+              if: 'true',
+              children: [
+                {
+                  $el: 'span',
+                  attrs: {
+                    class: '$classes.inputMaxlengthRemaining',
+                  },
+                  children: [
+                    `$maxlengthStringCount`,
+                    {
+                      $el: 'span',
+                      children: `$maxlengthStringCount + ' ' + $maxlengthRemainingText`,
+                      class: '$classes.inputMaxlengthRemainingHover',
+                    }
+                  ]
+                },
+                "$suffix || ''"
+              ]
+            }
+
+            return originalSchema(extensions)
+          }
+
+          inputDefinition.schema = higherOrderSchema
+          if (inputDefinition.schemaMemoKey) {
+            inputDefinition.schemaMemoKey += '-maxlength-remaining'
+          }
+          node.props.definition = inputDefinition
+
+          updateCountValue({ payload: node._value as string })
+          node.on('input', updateCountValue)
+
+          function updateCountValue ({ payload }: { payload: string }) {
+            let stringLength = parseInt(node.props.attrs.maxlength) - (payload ? payload.length : 0)
+            node.props.maxlengthStringCount = Math.max(0, stringLength)
+            node.props.maxlengthRemainingText = node.props.maxlengthRemainingText ||
+              MaxlengthCountdownOptions?.remainingText ||
+              'remaining characters'
+          }
+        }
```

Comment:
The same memoization technique discussed in the review of #1000 could apply here too. Let’s do as little work inside this input event as possible.